test/URLEndpointListenerTest.cc

```diff
@@ -397,4 +440,86 @@ TEST_CASE_METHOD(URLEndpointListenerTest, "Listener with Cert Authentication", "
         CBLTLSIdentity_Release(listenerConfig.tlsIdentity);
 }
 
+#ifdef __APPLE__
+TEST_CASE_METHOD(URLEndpointListenerTest, "Listener with Cert Authentication with External KeyPair", "[URLListener]") {
+    constexpr bool withExternalKey = true;
+
+    struct Context {
+        int rand = 6801;
+    } context;
+    
+    CBLURLEndpointListenerConfiguration listenerConfig {
+        &context, // context
+        cy.data(),
+        2,
+        0         // port
+    };
+    listenerConfig.disableTLS = false;
+
+    bool anonymous = false;
+    alloc_slice persistentLabel;
+
+    SECTION("Self-signed Cert") {
+        listenerConfig.tlsIdentity = createTLSIdentity(true, withExternalKey);
+    }
+
+    listenerConfig.authenticator = CBLListenerAuth_CreateCertificate([](void* ctx, FLSlice certData) {
+        auto context = reinterpret_cast<Context*>(ctx);
+        CHECK(context->rand  == 6801);
+        CBLError error;
+        CBLCert* cert = CBLCert_CreateWithData(certData, &error);
+        if (!cert) {
+            CBL_Log(kCBLLogDomainReplicator, kCBLLogError,
+                    "CBLCert_CertFromData failed with code %d", error.code);
+            return false;
+        }
+        alloc_slice sname = CBLCert_SubjectName(cert);
+        CBLCert_Release(cert);
+        return sname == slice("CN=URLEndpointListener_Client");
+    });
+    config.acceptOnlySelfSignedServerCertificate = true;
+    expectedDocumentCount = 20;
+
+    REQUIRE(listenerConfig.authenticator);
+    
+    createNumberedDocsWithPrefix(cx[0], 10, "doc");
+    createNumberedDocsWithPrefix(cx[1], 10, "doc");
+    createNumberedDocsWithPrefix(cy[0], 20, "doc2");
+    createNumberedDocsWithPrefix(cy[1], 20, "doc2");
+    auto cols = collectionConfigs({cx[0], cx[1]});
+    config.collections = cols.data();
+    config.collectionCount = cols.size();
+    CBLError error {};
+    CBLURLEndpointListener* listener = CBLURLEndpointListener_Create(&listenerConfig, &error);
+    REQUIRE(listener);
+    
+    CHECK(CBLURLEndpointListener_Start(listener, &error));
+
+    config.endpoint = clientEndpoint(listener, &error);
+    REQUIRE(config.endpoint);
+
+    CBLTLSIdentity* clientIdentity = createTLSIdentity(false, withExternalKey);
+    REQUIRE(clientIdentity);
+    config.authenticator = CBLAuth_CreateCertificate(clientIdentity);
+    REQUIRE(config.authenticator);
+    config.replicatorType = kCBLReplicatorTypePush;
+    replicate();
+
+    if (anonymous) {
```

Comment:
I think this is not relevant to this test as this test is testing with the external key callbacks?